### PR TITLE
Set sensible cookie flags on the admin app

### PR DESCRIPTION
**Pull request by nextgens**
## What type of PR?

Bugfix

## What does this PR do?

It sets the right flags on the session cookie issued by the admin app.
This should probably be backported as the lack of secure flag on TLS-enabled setup is a high risk vulnerability.

SameSite is hardening / helps against CSRF on modern browsers
HTTPOnly is hardening / helps reduce the impact of XSS

### CODE REVIEW

File: core/admin/mailu/configuration.py [hunk 1]
```diff
@@ -123,6 +123,9 @@ def init_app(self, app):
 
         self.config['RATELIMIT_STORAGE_URL'] = 'redis://{0}/2'.format(self.config['REDIS_ADDRESS'])
         self.config['QUOTA_STORAGE_URL'] = 'redis://{0}/1'.format(self.config['REDIS_ADDRESS'])
+        self.config['SESSION_COOKIE_SAMESITE'] = 'Strict'
+        self.config['SESSION_COOKIE_HTTPONLY'] = True
+        self.config['SESSION_COOKIE_SECURE'] = self.config['TLS_FLAVOR'] != 'notls'
```
Comment (lub) [inline core/admin/mailu/configuration.py:128]: I don't think that check is correct. For example if someone uses mail or mail-letsencrypt they don't necessarily access admin via https. The other way around when someone uses cert or letsencrypt they could still access admin unencrypted by exposing it directly (iirc setup does this by default).

Not sure what the proper solution would be. Adding x-forwarded-proto to nginx and checking for that? But that has other caveats, e.g. we'd have to check if the request originated from nginx.

Comment (nextgens) [reply core/admin/mailu/configuration.py:128]: I guess this is related to #1740
I'm not sure what the proper solution is either.

Maybe we should go further and delegate all of this to something like flask-talisman ?

Comment (nextgens) [reply core/admin/mailu/configuration.py:128]: One of the complications is that we don't have a flask request context at configuration time...

I'm not convinced that auto-detecting the right value at runtime is the way forward (hard to cover all the usecases, will be difficult to debug by users, ...). What about just documenting it and allowing the default to be overridden in the configuration?

Comment (nextgens) [reply core/admin/mailu/configuration.py:128]: Meh. The complexity for handling this is just too high.

https://mailu.io/1.7/general.html#philosophy
"Mailu has a central front container that routes all HTTP and mail traffic."

Can't we just stick with that? :)
And if we do, would you rather I get nginx to set the cookie flags? (set_cookie_flag https://docs.nginx.com/nginx/admin-guide/dynamic-modules/cookie-flag/)

Comment (nextgens) [reply core/admin/mailu/configuration.py]: I've implemented what was agreed on matrix in 0dcc059

**Review by lub — APPROVED** [review]

**Review by Diman0 — APPROVED** [review]
Lgtm. Since this is a security fix, i agree we should backport it.

**Comment by Diman0** [comment]
bors retry

**Pull request merged**